contracts/GaslessV3.sol

```diff
@@ -311,20 +333,42 @@ contract GaslessV3 is Ownable {
         );
 
         console.log('Fees: ', fees);
-        ERC20Permit token = ERC20Permit(params.tokenAddress);
-        token.permit(
-            params.userAddress,
-            address(this),
-            params.approvalValue,
-            params.approvalDeadline,
-            params.approvalSigV,
-            params.approvalSigR,
-            params.approvalSigS
-        );
-        require(
-            token.transferFrom(params.userAddress, address(this), fees),
-            '[APPROVE WITHOUT FEES] Failed to transfer from'
+        ERC20PermitUpgradeable token = ERC20PermitUpgradeable(
+            params.tokenAddress
         );
+        console.log('ok1');
+
+        if (params.tokenAddress == DAI_TOKEN_ADDRESS) {
```

Comment:
let's also add a chain id check for this

Reply:
it's already checking ethereum chain because every token has different address on different chain

Reply:
it's possible that the address is same across chains so as a safety let's add the check

Reply:
done, added the check